File: rsa-bruteforce.py
import subprocess
import sys

def cmdline(command) -> bytes:
    proc = subprocess.Popen(str(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True) 
    (out, err) = proc.communicate()
    return err

def main(file) -> None:
    count = 0
    # Iterate through the lines in the file and try each one as the password
    with open('rockyou.txt', 'r', encoding='koi8_u') as f:
        for w in f:
            w = w.strip()  # strip the leading/trailing whitespace
            # command for cli
            strcmd = f"openssl rsa -in {file} -out out.key -passin pass:{w}"
            res = cmdline(strcmd)
            count += 1
            if res.startswith(b"writing"):
                # working password is printed
                print("\nThe key is: " + w)
                sys.exit()
            else:
                print(f"{count} : {w}")
    print("Finished Wordlist")

if __name__ == '__main__':
    file = "private.pem"
    main(file)

File: rsa-bruteforce.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define BUFFER_SIZE 128

int try_password(const char *password) {
  char command[BUFFER_SIZE];
  sprintf(command, "openssl rsa -in private.pem -out out.key -passin pass:%s", password);
  int result = system(command);
  return result == 0;
}

int main(int argc, char *argv[]) {
  FILE *fp = fopen("rockyou.txt", "r");
  if (fp == NULL) {
    perror("Error opening file");
    return 1;
  }

  char password[BUFFER_SIZE];
  int count = 0;
  while (fgets(password, BUFFER_SIZE, fp) != NULL) {
    // Strip the newline character from the password
    password[strcspn(password, "\n")] = '\0';
    if (try_password(password)) {
      printf("Found password: %s\n", password);
      break;
    } else {
      count++;
      printf("Incorrect password: %s\n", password);
    }
  }

  fclose(fp);
  return 0;
}
